v3/mysql/mysql.go

```diff
@@ -0,0 +1,342 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"errors"
+	"io"
+	"reflect"
+
+	spindb "github.com/spinframework/spin-go-sdk/v3/internal/db"
+	"github.com/spinframework/spin-go-sdk/v3/internal/fermyon/spin/v2.0.0/mysql"
+	rdbmstypes "github.com/spinframework/spin-go-sdk/v3/internal/fermyon/spin/v2.0.0/rdbms-types"
+	"go.bytecodealliance.org/cm"
+)
+
+// Open returns a new connection to the database.
+func Open(name string) *sql.DB {
+	return sql.OpenDB(&connector{name: name})
+}
+
+type conn struct {
+	spinConn mysql.Connection
+}
+
+func (c *conn) Close() error {
+	return nil
+}
+
+func (c *conn) Prepare(query string) (driver.Stmt, error) {
+	return &stmt{conn: c, query: query}, nil
+}
+
+func (c *conn) Begin() (driver.Tx, error) {
+	return nil, errors.New("transactions are unsupported by this driver")
+}
+
+type connector struct {
+	conn *conn
+	name string
+}
+
+func (d *connector) Connect(_ context.Context) (driver.Conn, error) {
+	if d.conn != nil {
+		return d.conn, nil
+	}
+	return d.Open(d.name)
+}
+
+func (d *connector) Driver() driver.Driver {
+	return d
+}
+
+func (d *connector) Open(name string) (driver.Conn, error) {
+	results := mysql.ConnectionOpen(name)
+	if results.IsErr() {
+		return nil, toError(results.Err())
+	}
+	d.conn = &conn{spinConn: *results.OK()}
+	return d.conn, nil
+}
+
+func (d *connector) Close() error {
+	if d.conn != nil {
+		d.conn.Close()
+	}
+	return nil
+}
+
+type rows struct {
+	columns    []string
+	columnType []uint8
+	pos        int
+	len        int
```

Comment:
Suggested field rename

```suggestion
	numRows        int
```